### PR TITLE
 Added support for Metric Charts in alerts

### PR DESCRIPTION
**Pull request by EranGon**

### CODE REVIEW

File: src/SDK/SmartDetectorsSDK/AlertPresentation/ThresholdType.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.Azure.Monitoring.SmartDetectors.AlertPresentation
+{
+    /// <summary>
+    /// Enumeration of possible types of a metric chart threshold.
+    /// </summary>
+    public enum ThresholdType
```
Comment (liadtal) [inline src/SDK/SmartDetectorsSDK/AlertPresentation/ThresholdType.cs]: >ThresholdType [](start = 16, length = 13)

There should also be "Both"

File: src/SDK/SmartDetectorsSDK/AlertPresentation/MetricChart.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    /// </summary>
+    public class MetricChart
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MetricChart"/> class.
+        /// </summary>
+        /// <param name="metricName">The name of the metric to display in the chart.</param>
+        /// <param name="timeGrain">The time grain (resolution) of the metric to display.</param>
+        /// <param name="aggregationType">The aggregation type to use when displaying the metric.</param>
+        public MetricChart(
```
Comment (liadtal) [inline src/SDK/SmartDetectorsSDK/AlertPresentation/MetricChart.cs]: >MetricChart [](start = 15, length = 11)

Why not initialize everything in the ctor?

File: src/SDK/SmartDetectorsSDK/AlertPresentation/ThresholdType.cs (hunk 1 above)
Comment (EranGon) [reply src/SDK/SmartDetectorsSDK/AlertPresentation/ThresholdType.cs]: when it's added to the contracts, i'll add it here

---
In reply to: [258925051](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/52#discussion_r258925051) [](ancestors = 258925051)

File: src/SDK/SmartDetectorsSDK/AlertPresentation/MetricChart.cs (hunk 2 above)
Comment (EranGon) [reply src/SDK/SmartDetectorsSDK/AlertPresentation/MetricChart.cs]: i'm only initializing mandatory members here, to make life easier for the developer

---
In reply to: [258926689](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/52#discussion_r258926689) [](ancestors = 258926689)

**Pull request merged**